tests/api/v1/datadog/scenarios_test.go

```
@@ -0,0 +1,87 @@
+package test
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/DataDog/datadog-api-client-go/api/v1/datadog"
+	"github.com/DataDog/datadog-api-client-go/tests"
+	"github.com/go-bdd/gobdd"
+)
+
+func TestScenarios(t *testing.T) {
+	s := gobdd.NewSuite(
+		t,
+		gobdd.WithIgnoredTags([]string{"@todo"}),
+		gobdd.WithBeforeScenario(func(ctx gobdd.Context) {
+			client := datadog.NewAPIClient(NewConfiguration())
+			tests.SetCtx(ctx, context.WithValue(context.WithValue(
+				context.Background(),
+				datadog.ContextAPIKeys,
+				map[string]datadog.APIKey{},
+			), clientKey, client))
+			tests.SetData(ctx, make(map[string]interface{}))
+			tests.SetCleanup(ctx, make(map[string]func()))
+		}), gobdd.WithAfterScenario(func(ctx gobdd.Context) {
+			tests.RunCleanup(ctx)
+		}),
+	)
+	tests.ConfigureSteps(s)
+
+	s.AddStep(`a valid "apiKeyAuth" key in the system`, aValidAPIKeyAuth)
+	s.AddStep(`a valid "appKeyAuth" key in the system`, aValidAppKeyAuth)
+	s.AddStep(`an instance of "([^"]+)" API`, anInstanceOf)
+	s.Run()
+}
+
+func aValidAPIKeyAuth(t gobdd.StepTest, ctx gobdd.Context) {
+	if keys, ok := tests.GetCtx(ctx).Value(datadog.ContextAPIKeys).(map[string]datadog.APIKey); ok {
+		keys["apiKeyAuth"] = datadog.APIKey{
+			Key: os.Getenv("DD_TEST_CLIENT_API_KEY"),
```

We should maybe try and fail fast here if this environment variable isn't set.